File: providers/deepseek.py
"""
DeepSeek 余额查询适配器
"""
from .base import ProviderSpec, SimpleHTTPProvider


def _extract(data):
    """余额在 balance_infos 列表里，优先取人民币账户，没有就退回第一条"""
    infos = data.get('balance_infos')
    if not isinstance(infos, list) or not infos:
        raise ValueError("无法从响应中解析 balance_infos 字段")

    chosen = next((i for i in infos if isinstance(i, dict) and i.get('currency') == 'CNY'), infos[0])
    total = chosen.get('total_balance') if isinstance(chosen, dict) else None
    if total is None:
        raise ValueError("无法从响应中解析 total_balance 字段")
    return float(total)


class DeepSeekProvider(SimpleHTTPProvider):
    """DeepSeek 服务商适配器"""

    SPEC = ProviderSpec(
        name="DeepSeek",
        url="https://api.deepseek.com/user/balance",
        headers={"Accept": "application/json"},
        extract=_extract,
    )

File: internal/provider/deepseek.go
package provider

import "errors"

// Implementation note.
var deepseekSpec = Spec{
	Key:         "deepseek",
	Name:        "DeepSeek",
	DefaultType: "balance",
	URL:         "https://api.deepseek.com/user/balance",
	Headers:     map[string]string{"Accept": "application/json"},
	Extract: func(data map[string]any) (float64, error) {
		infos, ok := data["balance_infos"].([]any)
		if !ok || len(infos) == 0 {
			return 0, errors.New("Could not parse balance_infos field")
		}

		// Implementation note.
		chosen := Object(infos[0])
		for _, item := range infos {
			if obj := Object(item); obj != nil && Str(obj["currency"]) == "CNY" {
				chosen = obj
				break
			}
		}

		// Implementation note.
		total, ok := Num(chosen["total_balance"])
		if !ok {
			return 0, errors.New("Could not parse total_balance field")
		}
		return total, nil
	},
}

func init() { RegisterSpec(deepseekSpec) }
